rs/registry/canister/canister/canister.rs

```diff
@@ -186,34 +185,54 @@ ic_nervous_system_common_build_metadata::define_get_build_metadata_candid_method
 
 #[export_name = "canister_query get_changes_since"]
 fn get_changes_since() {
-    let response_pb = match deserialize_get_changes_since_request(arg_data()) {
-        Ok(version) => {
-            let registry = registry();
+    fn main() -> Result<RegistryGetChangesSinceResponse, (Code, String)> {
+        // Canisters are not allowed to call us.
```

Comment:
Could we check who is _allowed_ to call us? For now, that would be any self-authenticating principal with the possibility to restrict the check later.

Reply:
I'm fine either way. The ticket said to block canisters, so that's what I implemented.

Reply:
Let's then please switch to an explicit list of allowed callers: self-authenticating and anonymous for now.